browser/ui/whats_new/whats_new_util.cc

```diff
@@ -0,0 +1,142 @@
+  }
+
+  const auto& version = version_info::GetVersion();
+  DCHECK(version.IsValid());
+  DCHECK_EQ(version.components().size(), 4ul);
+
+  // |version| has four components like 111.1.51.34.
+  // First one is upstream's major version.
+  // Brave's major version is second and third component - 1.51
+  // Ignored fourth number as it's build number.
+  double current_version;
+  if (!base::StringToDouble(base::StringPrintf("%d.%d", version.components()[1],
+                                               version.components()[2]),
+                            &current_version)) {
+    return absl::nullopt;
+  }
+
+  return current_version;
+}
+
+}  // namespace
+
+namespace whats_new {
+
+void SetCurrentVersionForTesting(double major_version) {
+  g_testing_major_version = major_version;
+}
+
+bool ShouldShowBraveWhatsNewForState(PrefService* local_state) {
+  // Supported languages/translations for the What's New page are:
+  // Simplified Chinese, French, German, Japanese, Korean, Portuguese, and
+  // Spanish.
+  constexpr std::array<const char*, 8> kSupportedLanguages = {
+      "en", "zh", "fr", "de", "ja", "ko", "pt", "es"};
+  const std::string default_lang_code =
+      brave_l10n::GetDefaultISOLanguageCodeString();
+  if (base::ranges::find(kSupportedLanguages, default_lang_code) ==
+      std::end(kSupportedLanguages)) {
+    VLOG(2) << __func__ << " Not supported language - " << default_lang_code;
+    return false;
+  }
+
+  auto target_major_version = GetTargetMajorVersion();
+  // false if whatsnew is not supported in this country.
+  if (!target_major_version) {
+    VLOG(2) << __func__ << " Field trial doesn't have target_major_version";
+    return false;
+  }
+
+  const auto current_version = GetCurrentBrowserVersion();
+  if (!current_version) {
+    NOTREACHED() << __func__ << " Should get current version.";
+    return false;
+  }
+
+  if (*current_version != *target_major_version) {
+    VLOG(2) << __func__ << " Current version is different with target version";
+    VLOG(2) << __func__ << " Current version - " << *current_version;
+    VLOG(2) << __func__ << " Target version - " << *target_major_version;
+    return false;
+  }
+
+  // Already shown whatsnew.
+  const double last_version =
+      local_state->GetDouble(prefs::kWhatsNewLastVersion);
+  if (last_version == *target_major_version) {
+    VLOG(2) << __func__ << " Already shown for " << *target_major_version;
+    return false;
+  }
+
+  // Set the last version here to indicate that What's New should not attempt
+  // to display again for this milestone. This prevents the page from
+  // potentially displaying multiple times in a given milestone, e.g. for
+  // multiple profile relaunches (see https://crbug.com/1274313).
+  local_state->SetDouble(prefs::kWhatsNewLastVersion, *target_major_version);
+  return true;
+}
+
+void RegisterLocalStatePrefs(PrefRegistrySimple* registry) {
+  registry->RegisterDoublePref(prefs::kWhatsNewLastVersion, 0);
```

Comment:
why are we creating our pref for this instead of using upstream `kLastWhatsNewVersion`?